src/core/Toast/Toast.tsx

```diff
@@ -150,32 +149,64 @@ export const Toast = (props: ToastProps) => {
           }}
         >
           <div ref={onRef}>
-            <div className={`iui-toast iui-${category}`}>
-              <div className='iui-status-area'>
-                {<StatusIcon className='iui-icon' />}
-              </div>
-              <div className='iui-message'>{content}</div>
-              {link && (
-                <a className='iui-anchor' onClick={link.onClick}>
-                  {link.title}
-                </a>
-              )}
-              {(type === 'persisting' || hasCloseButton) && (
-                <IconButton
-                  size='small'
-                  styleType='borderless'
-                  onClick={close}
-                  aria-label='Close'
-                >
-                  <SvgCloseSmall />
-                </IconButton>
-              )}
-            </div>
+            {ToastPresentation({
+              category,
+              content,
+              link,
+              type,
+              hasCloseButton,
+              onRemove: close,
+            })}
           </div>
         </div>
       )}
     </Transition>
   );
 };
 
+/**
+ * The presentational part of a toast notification, without any animation or logic.
+ * @private
+ */
+export const ToastPresentation = (
+  props: Omit<ToastProps, 'duration' | 'id' | 'isVisible'> & CommonProps,
+) => {
+  const {
+    content,
+    category,
+    type = 'temporary',
+    link,
+    hasCloseButton,
+    onRemove,
```

Comment:
hmm.. discussable.
I would say it onClose then (cause its called only when you click the close 'x' button)

Reply:
Changed to `onClose`. I didn't want to declare new props for this internal component but I guess it's fine 🤷‍♀️